File: app/src/models/Course.ts
export type Course = {
    id: number
    title: string
    description: string
    theme: string
    author: string
    authorId: number
    techstack: Array<string>
    syllabus: Array<string>
    beneficiars: Array<string>
    duration: string
    difficulty: string
    createdOn: string
    updatedOn: string
}

File: api/pkg/courses/course.go
package courses

// Course defines the structure for an API
type Course struct {
	ID          string   `json:"id"`
	Title       string   `json:"title" validate:"required"`
	Description string   `json:"description" validate:"required"`
	Theme       string   `json:"theme"`
	Author      string   `json:"author" validate:"required,author"` // TODO author is to be Autor struct
	AuthorID    string   `json:"authorid"`
	TechStack   []string `json:"techstack"`
	Syllabus    []string `json:"syllabus"`
	Duration    string   `json:"-"`
	Beneficiars []string `json:"beneficiars"`
	Difficulty  string   `json:"-"`
	CreatedOn   string   `json:"-"`
	UpdatedOn   string   `json:"-"`
}
